celltype_annotation/analysis/cell-assign-sarcoma-marker-genes.Rmd

```diff
@@ -0,0 +1,194 @@
+  anndata_dir, 
+  glue::glue("{params$library_id}_rms_broad_markers.tsv")
+)
+
+```
+
+```{r}
+# read in all annotated sce 
+processed_sce <- readr::read_rds(local_processed_sce_path)
+```
+
+```{r}
+# source helper functions
+function_file <- file.path("..", "utils", "cellassign-helper-functions.R")
+source(function_file)
+```
+
+## Get marker genes
+
+```{r}
+# extract coldata for easy plotting
+coldata_df <- colData(processed_sce) |>
+  as.data.frame() |> 
+  dplyr::select(barcode, celltype) |>
+  # classify cells as tumor or normal
+  dplyr::mutate(cell_category = dplyr::if_else(stringr::str_detect(celltype, "Tumor"), "Tumor", celltype),
+                # remove sub states of myoblast/mesoderm/myocyte
+                broad_celltype = stringr::str_remove(celltype, "-A|-B|-C|-D"))
+```
+
+
+```{r}
+# find markers for all cell types 
+processed_sce_markers <- scran::findMarkers(processed_sce,
+                                            groups = coldata_df$broad_celltype,
+                                            pval.type = "all")
+```
+
+
+```{r}
+# get top 20 markers for each cell type 
+top_markers_df <- as.list(processed_sce_markers) |>
+  purrr::imap(
+    \(markers, id) 
+    markers |> 
+      as.data.frame() |>
+      dplyr::arrange(desc(summary.logFC)) |>
+      dplyr::mutate(celltype = id) |>
+      tibble::rownames_to_column("ensembl_id") |> 
+      dplyr::select(celltype, ensembl_id) |>
+      head(n = 10)
+  ) |>
+  dplyr::bind_rows()
+
+# create a binary matrix of genes by cell type markers 
+markers_binary_mtx <- top_markers_df |>
+  dplyr::select(celltype, ensembl_id) |> 
+  tidyr::pivot_wider(ensembl_id,
+                     names_from = celltype,
+                     values_from = celltype,
+                     values_fn = length,
+                     values_fill = 0) |> 
+  tibble::column_to_rownames("ensembl_id") |>
+  dplyr::mutate(other = 0)
+
+# replace length with 1
+markers_binary_mtx[markers_binary_mtx > 1] <- 1
+
+# add ensembl id as first column
+markers_binary_mtx <- markers_binary_mtx |>
+  tibble::rownames_to_column("ensembl_id") |>
+  dplyr::relocate(ensembl_id)
+```
+
+```{r}
+# look at myoblast markers, are they what we expect? 
+myoblast_markers <- top_markers_df |>
+  dplyr::filter(celltype == "Tumor_Myoblast")
+
+# do some mapping to get the gene symbols
+myoblast_markers <- AnnotationDbi::mapIds(
+    # organism annotation package
+    org.Hs.eg.db::org.Hs.eg.db,
+    # the provided gene identifiers
+    keys = myoblast_markers$ensembl_id,
+    # the type of provided gene identifiers
+    keytype = "ENSEMBL",
+    # the type of gene identifiers to map to
+    column = "SYMBOL"
+  ) |>
+    tibble::enframe(name = 'ensembl_id',
+                    value = tolower("SYMBOL")) |> 
+    dplyr::left_join(myoblast_markers, by = "ensembl_id")
+
+myoblast_markers
+```
+I kind of would expect some of the myoblast markers to be included here like `MYOG`, `MYF5`, `MYOD1`, etc. 
```

Comment:
Are they not present because those would also be present in non-tumor myoblast cells? Are those among the cells annotated here? If so, we might expect to recover those if using `pval.type = "any"` in `findMarkers()`.

(My thought earlier was to use some combination of the markers found by `any` and `all` to supply to CellAssign)